README.md

```diff
@@ -1 +1,152 @@
+* T. A. White, R. A. Kirian, A. V. Martin, A. Aquila, K. Nass, A. Barty and H. N. Chapman. "CrystFEL: a software suite for snapshot serial crystallography". J. Appl. Cryst. 45 (2012), p335–341.
+* R. A. Kirian, X. Wang, U. Weierstall, K. E. Schmidt, J. C. H. Spence, M. Hunter, P. Fromme, T. White, H. N. Chapman and J. Holton. "Femtosecond protein nanocrystallography - data analysis methods". Optics Express 18 (2010) p5713. 
+
+
+
+
+## Tools: 
+
+* hdfsee.py - for displaying a diffraction image from a h5 file.
+
+* GUI_tools.py - displaying histograms from a indexing.stream file (produced by indexamajig) and fitting gaussian curves.
+  
+
+  
+
+## Dependencies:
+
+```CrystFEL``` (http://www.desy.de/~twhite/crystfel/install.html)
+
+```Python 3 environment```
+ 
+```matplotlib```
+
+```jupyter notebook```
+
+```bash kernel```
+
+```cfelpyutils```
+
+```ccp4```
+
+```dirax```
+
+```hdf5```
+
+```xds```
+
+
+### Installing bash kernel:
+```git clone https://github.com/takluyver/bash_kernel.git```
+
+```pip install bash_kernel```
+
+```python -m bash_kernel.install```
+
+ 
+ ## Tests
+
+`python -m unittest discover -s tests/ -p 'unit_*' -v`
+
+  
+  
+
+## hdfsee.py usage:
+
+  
+
+The tool is designed to work with h5 files (hdf5). Tested only with LCLS specific data structure (http://www.cxidb.org/data/21/cxidb-21-run0130.tar).
+Features buttons for changing contrast and colour space, displaying diffraction peaks from indexing.stream file or peaks embedded in the h5 file.
+
+ 
+### Example usage:
+  
+
+  
+
+1. Basic displaynig of the data as it is:
+
+  
+
+```python hdfsee.py <filename>```
+
+  
+
+2. Displaying the image with geometry reconstruction:
+  
+
+```python hdfsee.py <filename> -g <geometry file>```
+
+  
+
+3. Displaying the image with geometry reconstruction and peaks written in the indexing.stream file produced by indexamajig:
+ 
+
+```python hdfsee.py <filename> -g <geometry file> -p <stream file>```
+
+
+
+#### Examples with files included in the repository:  
+  
+
+```python hdfsee.py LCLS_2013_Mar23_r0132_010310_df11.h5```
+
+  
+
+```python hdfsee.py LCLS_2013_Mar23_r0132_010310_df11.h5 -g 5HT2B-Liu-2013.geom```
+
+  
+  
+
+**check-peak-detection.sh**:
+
+
+```./check-peak-detection.sh <stream file> <geometry file>```
+
+  
+**Instructions for running on maxwell cluster**  
+
+#####TODO#####
+
+
+
+## GUI_tools.py usage:
+
+GUI_tools.py is a ```cell_explorer``` tool from CrystFEL ported to Python. 
+To select a range of intrerest for gaussian fitting press and hold left mouse button, 
+release the button after the selection is ready. To deselect click left mouse button 
+anywhere on the plot. Pressing '+' or '-' changes the binning size by a factor 
+of 2 (128 -> 64). Pressing 'g' key fitts the gaussian curve to the selected range 
+and displays fitting parameters.
+
+
+#### Example:
+
+
+```python GUI_tools.py Klaster.stream```
+
+#### Example in jupyter notebook:
+
+```magic-cell_explorer_and_hdfsee.ipynb```
+
+
```

Comment:
These points don't really matter because it's rendered the same in the end anyway. Some general comments on the markdown document:

1. Two a single back tick pair works fine single-line code snippet, no need for three:
`magic-cell_explorer_and_hdfsee.ipynb`

2. Paragraphs should have a single line break between them